src/test/java/com/github/stefanbirkner/systemlambda/WithTextFromSystemInTest.java

```diff
@@ -37,18 +37,19 @@ static void checkArrayConstants() {
 	@Test
 	void provided_text_is_available_from_system_in(
 	) throws Exception {
+		AtomicReference<String> firstLineCapture = new AtomicReference<>();
 		AtomicReference<String> secondLineCapture = new AtomicReference<>();
 
 		withTextFromSystemIn(
 			"first line",
 			"second line"
 		).execute(() -> {
-			Scanner firstScanner = new Scanner(in);
-			firstScanner.nextLine();
-			Scanner secondScanner = new Scanner(in);
```

Comment:
The first scanner may cache more than just the first line, so the second scanner is unable to pick up the second line.